Enhancements to VS solution generation

TechnoPorg opened the pull request:
Fixes #66008.
Adds support for the `dev_mode=yes` and `float=64` command line options when generating a Visual Studio solution.
This also includes the changes from #66718, so that solution generation works at all.
The whole solution generation system feels messy to me, but I'm not sure if there's a better way to do it.

<!--
Please target the `master` branch in priority.
PRs can target `3.x` if the same change was done in `master`, or is not relevant there.

Relevant fixes are cherry-picked for stable branches as needed by maintainers.
You can mention in the description if the change is compatible with `3.x`.
-->

akien-mga commented on methods.py:
```diff
@@ -742,10 +742,12 @@ class ModuleConfigs(Mapping):
             PLATFORM_IDS = ["x86_32", "x86_64"]
             CONFIGURATIONS = ["editor", "template_release", "template_debug"]
             DEV_SUFFIX = ["", ".dev"]
+            DOUBLE_SUFFIX = ["", ".double"]
```
I don't think this is needed, it's too niche and will mean that there's 24 (!) variants to choose from.

Handling the option in `build_commandline` like done for `tests` is enough IMO.

Commit:
Various enhancements to Visual Studio solution generation.

This adds support for building solutions with dev_mode and/or float=64 enabled.
Additionally, it adds solution generation to the Windows CI to catch future regressions.

TechnoPorg commented:
I have also added `vsproj=yes` to the Windows CI builds, in order to hopefully catch future regressions with solution building. I'm not sure if it will work properly in GitHub Actions, though.

akien-mga commented:
Thanks!
